#! /usr/bin/env python

PACKAGE='tod_video'

from dynamic_reconfigure.parameter_generator_catkin import *

class SensorLevels:
    RECONFIGURE_RUNNING = 0
    RECONFIGURE_STOP = 1

gen = ParameterGenerator()

scalings = gen.enum([
    gen.const("1p000", str_t, "1p000", ""),
    # gen.const("0p875", str_t, "0p875", ""),
    gen.const("0p750", str_t, "0p750", ""),
    # gen.const("0p675", str_t, "0p625", ""),
    gen.const("0p500", str_t, "0p500", ""),
    # gen.const("0p375", str_t, "0p375", ""),
    gen.const("0p250", str_t, "0p250", ""),
    gen.const("0p125", str_t, "0p125", ""), # still needed for rear camera which is not binned
    ], "Scalings.")

# Name    Type    Reconfiguration Level    Description    Default    Min    Max
gen.add("camera_name", str_t, SensorLevels.RECONFIGURE_RUNNING, "Name of Camera.", "")
gen.add("paused", bool_t, SensorLevels.RECONFIGURE_RUNNING, "store if stream is active", True)
gen.add("actual_width", int_t, SensorLevels.RECONFIGURE_RUNNING, "actual width.", -1, -1, 2000)
gen.add("actual_height", int_t, SensorLevels.RECONFIGURE_RUNNING, "actual height.", -1, -1, 2000)
gen.add("bitrate", int_t, SensorLevels.RECONFIGURE_RUNNING, "bitrate.", -1, -1, 10000)
gen.add("scaling", str_t, SensorLevels.RECONFIGURE_RUNNING, "scaling.", "1p000", edit_method = scalings)
gen.add("width", int_t, SensorLevels.RECONFIGURE_RUNNING, "width.", -1, -1, 2000)
gen.add("height", int_t, SensorLevels.RECONFIGURE_RUNNING, "height.", -1, -1, 2000)
gen.add("offset_width", int_t, SensorLevels.RECONFIGURE_RUNNING, "offset width.", -1, -1, 2000)
gen.add("offset_height", int_t, SensorLevels.RECONFIGURE_RUNNING, "offset height.", -1, -1, 2000)

exit(gen.generate(PACKAGE, "tod_video", "Video"))
